src/download.ts

```diff
@@ -41,13 +42,21 @@ export class Download {
     this._finishedPromise = new Promise(f => this._finishedCallback = f);
     page._browserContext._downloads.add(this);
     this._acceptDownloads = !!this._page._browserContext._options.acceptDownloads;
+  }
+
+  _filenameSuggested(suggestedFilename: string) {
+    this._suggestedFilename = suggestedFilename;
     this._page.emit(Events.Page.Download, this);
   }
 
   url(): string {
     return this._url;
   }
 
+  suggestedFilename(): string {
```

Comment:
Let's make this method async and fire `download` event on a page that is guaranteed to be alive as there is a chance that at least in webkit that suggested file name may come after the page has been closed.

Reply:
Can't do - the api becomes very ugly if I do.

Reply:
Can you elaborate ? In particular why it'd be worse compared to firing download event on a closed page?

Reply:
If we don't fire suggested file name in webkit, it means there is no download, so I think it is safe to not emit this even in case of such behavior. With that, I'd like to keep suggestedFilename sync to the ease of use.